src/vumi2/transports/smpp/client.py

```diff
@@ -198,4 +212,25 @@ async def send_pdu(self, pdu: Union[PDURequest, PDUResponse]) -> Optional[PDU]:
 
     async def send_vumi_message(self, message: Message):
         for pdu in await self.submit_sm_processor.handle_outbound_message(message):
-            await self.send_pdu(pdu)
+            response = await self.send_pdu(pdu, check_response=False)
```

Comment:
What if one of the messages other than the last one fails?

Reply:
So at the moment `send_pdu` will only ever send a single PDU. When we get to multipart, we'll have to deal with multiple PDUs, but at the moment I'm not bothering with it. A simple way would be if any of the PDUs get an error response, then stop and respond with a nack, but I need to look more closely at the multipart implementation first before I can know what to do.

Reply:
Maybe add a comment to that effect so we don't forget?